powershell/Tests/lib/Start-Engine.ps1

```diff
@@ -71,23 +82,26 @@ while ($true) {
 
     # Cleanup the environment variables.
     $currentMatches = @{ }
-    foreach ($envVar in (Get-ChildItem -LiteralPath env:)) {
+    foreach ($key in ([Environment]::GetEnvironmentVariables().Keys)) {
+        $value = [Environment]::GetEnvironmentVariable($key)
+
         # Remove the environment variable if it is new.
-        if (!$originalEnv.ContainsKey($envVar.Name)) {
-            Remove-Item -LiteralPath $envVar.PSPath
-        } elseif ($originalEnv[$envVar.Name] -ceq $envVar.Value) {
+        if (!$originalEnv.ContainsKey($key)) {
+            [Environment]::SetEnvironmentVariable($key, $null)
+        } elseif ($originalEnv[$key] -ceq $value) {
             # Otherwise record it if it matches.
-            $currentMatches[$envVar.Name] = $envVar.Value
+            $currentMatches[$key] = $true
         }
     }
 
     # Add or update the environment variables that are missing or changed.
     foreach ($key in $originalEnv.Keys) {
         if (!$currentMatches.ContainsKey($key)) {
-            Set-Content -LiteralPath "env:$key" -Value "$($originalEnv[$key])"
+            [Environment]::SetEnvironmentVariable($key, $originalEnv[$key])
         }
     }
 
     # Write a special "end-of-test" message over STDOUT.
     Write-Host '_END_OF_TEST_ce10a77a_'
+    [Console]::Out.Flush()
```

Comment:
this is a lesson learned from problems we had in the tasks repo. when the machine is under high load it appears powershell (or .net?) may delay the flush for a long while. this came up when the tasks were timing out in the tasks repo, and i had even bumped the timeouts to 2 minutes and still timing out - finally flush did the trick and i was able to lower the timeouts to something sane